tests/prometheus_test.go

```diff
@@ -283,14 +283,22 @@ func getPrometheusSaToken(k8sClient *kubernetes.Clientset) (string, error) {
 			secretName = secret.Name
 		}
 	}
-	secret, err := k8sClient.CoreV1().Secrets(monitoringNs).Get(context.TODO(), secretName, metav1.GetOptions{})
-	if err != nil {
-		return "", err
-	}
-	if _, ok := secret.Data["token"]; !ok {
-		return "", fmt.Errorf("No token data in secret")
+	if secretName == "" {
+		// Since 1.24 SAs don't have tokens automatically generated, we determined the SA has no secret, so we
```

Comment:
if you want to use the API directly you can check out what we did in CDI
https://github.com/kubevirt/containerized-data-importer/pull/2328/commits/79c8c331eb59195ffad6a30f30394e9807162833

Reply:
Thanks, I will fix that, looks much better than me calling kubectl.